src/IonBinaryWriter.ts

```diff
@@ -221,7 +221,7 @@ export class BinaryWriter extends AbstractWriter {
 
   writeSymbol(value: string) : void {
     this.checkWriteValue();
-    if (value === null) {
+    if (value === null || value === undefined) {
```

Comment:
do we want to allow undefined input here?

Reply:
> do we want to allow undefined input here? I personally would rather throw.

Me too. I've opened [an issue](https://github.com/amzn/ion-js/issues/456) to discuss and track this change.

In the meantime, `writeSymbol` is the only odd-man-out that doesn't accept `undefined`. This required a test to be skipped, so I addressed it here. We can change the behavior across the board all in a future PR.